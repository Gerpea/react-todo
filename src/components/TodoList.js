import React from 'react'
import PropTypes from 'prop-types'

import TodoItem from './TodoItem'
import { Button } from 'react-bootstrap'

const TodoList = ({ todos }) => {
  return (
    <>
      <div data-testid='todo-list'>
        {todos.map((todo, i) => {
          return <TodoItem isDone={todo.isDone} title={todo.title} key={todo.uid ?? i} />
        })}
      </div>
      <Button data-testid='add-item' variant='primary'>
        Добавить элемент
      </Button>
    </>
  )
}

TodoList.propTypes = {
  todos: PropTypes.array,
}

TodoList.defaultProps = {
  todos: [],
}

export default TodoList
