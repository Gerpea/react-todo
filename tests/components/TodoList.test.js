import React from 'react'
import ReactDOM from 'react-dom'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

import TodoList from '../../src/components/TodoList'

afterEach(cleanup)

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<TodoList />, div)
})

describe('todos', () => {
  it('should render empty list by default', () => {
    const { getByTestId } = render(<TodoList />)

    expect(getByTestId('todo-list').hasChildNodes()).toBeFalsy()
  })

  it('should render correct number of todos', () => {
    const todos = [{}, {}, {}]

    const { getByTestId } = render(<TodoList todos={todos} />)

    expect(getByTestId('todo-list').childElementCount).toBe(todos.length)
  })

  it('should renders correctly', () => {
    const todo = {
      title: 'Item',
    }

    const { queryByText } = render(<TodoList todos={[todo]} />)

    expect(queryByText(todo.title)).toBeInTheDocument()
  })
})

describe('add item', () => {
  it('should render button corectly', () => {
    const { queryByTestId } = render(<TodoList />)

    expect(queryByTestId('add-item')).not.toBeNull()
  })
})

it('matches snapshot', () => {
  const tree = renderer.create(<TodoList />).toJSON()

  expect(tree).toMatchSnapshot()
})
