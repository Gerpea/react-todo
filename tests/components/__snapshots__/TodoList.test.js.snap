// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    data-testid="todo-list"
  />,
  <button
    className="btn btn-primary"
    data-testid="add-item"
    disabled={false}
    type="button"
  >
    Добавить элемент
  </button>,
]
`;
